addon/-private/system/references/has-many.js

```diff
@@ -121,24 +121,26 @@ HasManyReference.prototype._isLoaded = function() {
 };
 
 HasManyReference.prototype.value = function() {
-  if (this._isLoaded()) {
+  if (this.isLoaded()) {
     return this.hasManyRelationship.manyArray;
   }
 
   return null;
 };
 
 HasManyReference.prototype.load = function() {
-  if (!this._isLoaded()) {
-    return this.hasManyRelationship.getRecords();
+  let manyArray = this.value();
+
+  if (!manyArray) {
+    manyArray = this.hasManyRelationship.getRecords();
   }
 
-  var manyArray = this.hasManyRelationship.manyArray;
   return Ember.RSVP.resolve(manyArray);
 };
 
 HasManyReference.prototype.reload = function() {
-  return this.hasManyRelationship.reload();
+  return this.hasManyRelationship.reload()
+    .then(() => this.value());
```

Comment:
And it will reject with errors. I am only overriding resolved result. I added some tests to prouve me right :)